### PR TITLE
fix: avoid false settings warnings on world hops

### PR DESCRIPTION
**Pull request by iProdigy**
Supersedes #127

### CODE REVIEW

**Review by Felanbird — APPROVED** [review]

File: src/main/java/dinkplugin/SettingsManager.java [hunk 1]
```diff
@@ -181,9 +188,50 @@ void onConfigChanged(ConfigChanged event) {
         }
     }
 
+    void onGameState(GameStateChanged event) {
+        if (event.getGameState() != GameState.LOGGED_IN) {
+            justLoggedIn.set(false);
+            return;
+        } else {
+            justLoggedIn.set(true);
+        }
```
Comment (iProdigy) [inline src/main/java/dinkplugin/SettingsManager.java:197]: Does this improve readability?

```suggestion
        boolean loggedIn = event.getGameState() == GameState.LOGGED_IN;
        justLoggedIn.set(loggedIn);
        if (!loggedIn) return;
```

File: src/main/java/dinkplugin/SettingsManager.java [hunk 2]
```diff
@@ -181,9 +188,50 @@ void onConfigChanged(ConfigChanged event) {
...
+            if (justLoggedIn.get())
+                return false; // try again on next tick
+
+            if (config.notifyCollectionLog() && client.getVarbitValue(Varbits.COLLECTION_LOG_NOTIFICATION) == 0) {
+                warnForGameSetting(CollectionNotifier.ADDITION_WARNING);
+            }
+
+            if (config.notifyPet() && (client.getVarbitValue(PetNotifier.UNTRADEABLE_LOOT_DROPS) == 0 || client.getVarbitValue(PetNotifier.LOOT_DROP_NOTIFICATIONS) == 0)) {
+                warnForGameSetting(PetNotifier.UNTRADEABLE_WARNING);
+            }
```
Comment (iProdigy) [inline src/main/java/dinkplugin/SettingsManager.java:212]: May clarify the logic

```suggestion
            if (config.notifyCollectionLog() && client.getVarbitValue(Varbits.COLLECTION_LOG_NOTIFICATION) == 0) {
            assert isCollectionLogInvalid(0);
                warnForGameSetting(CollectionNotifier.ADDITION_WARNING);
            }

            if (config.notifyPet() && (client.getVarbitValue(PetNotifier.UNTRADEABLE_LOOT_DROPS) == 0 || client.getVarbitValue(PetNotifier.LOOT_DROP_NOTIFICATIONS) == 0)) {
            assert isPetLootInvalid(0);
                warnForGameSetting(PetNotifier.UNTRADEABLE_WARNING);
            }
```

File: src/main/java/dinkplugin/SettingsManager.java [hunk 3]
```diff
@@ -361,4 +411,14 @@ private void handleImport(Map<String, Object> map) {
...
     }
+
+    static {
+        PROBLEMATIC_VARBITS = ImmutableSet.of(
+            KillCountNotifier.KILL_COUNT_SPAM_FILTER,
+            CombatTaskNotifier.COMBAT_TASK_REPEAT_POPUP,
+            Varbits.COLLECTION_LOG_NOTIFICATION,
+            PetNotifier.LOOT_DROP_NOTIFICATIONS,
+            PetNotifier.UNTRADEABLE_LOOT_DROPS
+        );
```
Comment (iProdigy) [inline src/main/java/dinkplugin/SettingsManager.java:422]: This is only used for `contains` lookup; should we avoid `Set` overhead by just creating a helper method like so
```java
public static boolean isProblematicVarbit(int id) {
    return id == X || id == Y || id == Z;
}
```

Comment (pajlada) [reply src/main/java/dinkplugin/SettingsManager.java:422]: I think this solution is fine

File: src/main/java/dinkplugin/SettingsManager.java (hunk 1 above)
Comment (pajlada) [reply src/main/java/dinkplugin/SettingsManager.java:197]: I think this is readable enough already

File: src/main/java/dinkplugin/SettingsManager.java (hunk 2 above)
Comment (pajlada) [reply src/main/java/dinkplugin/SettingsManager.java:212]: Current code is fine imo

**Review by pajlada — APPROVED** [review]
I haven't tested this but readability-wise it looks good to me

**Pull request merged**